domain/bankaccount.py

```diff
@@ -30,26 +30,25 @@ def transfer(self, destiny_id: AccountID, value: Amount) -> None:
 
         main_balance = self._main_account.get_balance()
         main_id = self._main_account.account_id
-        if main_balance < value:
-            remaining = value - main_balance
-            draft_account_balance = self._draft_account.get_balance()
-            overdraft = self._overdraft_limit - draft_account_balance
-
-            if overdraft >= remaining:
-                draft_id = self._draft_account.account_id
-                draft_transaction = create_transaction(draft_id, main_id, remaining)
-                main_transaction = create_transaction(main_id, destiny_id, value)
-
-                self._draft_account.add_transaction(draft_transaction)
-                self._main_account.add_transaction(draft_transaction)
-                self._main_account.add_transaction(main_transaction)
-
-            else:
-                raise InsufficientFundsException(self.get_balance(), value)
-        else:
-
-            transaction = create_transaction(main_id, destiny_id, value)
-            self._main_account.transactions.insert(0, transaction)
+
+        main_transaction = create_transaction(main_id, destiny_id, value)
+        if main_balance >= value:
+            self._main_account.add_transaction(main_transaction)
```

Comment:
that line is equal to 51 line. The point of the refactor is to don't duplicate that lines, so we can extract a function to ask overdraft if needed